Automattic/woocommerce-payments
# Fix Printed receipt endpoint throws error when called from Jetpack API

Fixes #3900

#### Changes proposed in this Pull Request


Change the content type of the response of the endpoint to generate a printed receipt.

When the endpoint is called from the Jetpack API, it throws a 500 error because the API expects `application/json` type. 




<img width="1181" alt="Screen Shot 2022-03-04 at 10 59 58" src="https://user-images.githubusercontent.com/1553182/158595466-8f0c9571-e186-4d01-bdf5-1869035ea2f2.png">

#### Expected outcome
<img width="1265" alt="Screen Shot 2022-03-04 at 11 15 30" src="https://user-images.githubusercontent.com/1553182/158595611-7fdba006-93be-46df-8055-f8c6cf0ba463.png">



#### Testing instructions


1. Be sure that you have at least one IPP payment available in your test store. Follow instructions on https://github.com/Automattic/woocommerce-payments/pull/3433 to learn how to do it.
2. Make a GET request to `payments/readers/receipts/<payment_intent_id> ` through Jetpack API i.e. where `payment_intent_id` is the one you created in previous step. Please find [here](pdjVC1-o4-p2) instructions to use WP.COM API developer console to do it.
3. Check that there are no errors in the response.
4. Check that the printed receipt HTML is present in the json response.


-------------------


**Post merge**
